Bindings/Java/Matlab/Utilities/osimTableToStruct.m

```diff
@@ -77,14 +77,16 @@
     
     % Get the osim table column label
     col_label  = char(osimtable.getColumnLabels.get(iLabel));
-    
+    disp_label = col_label;
     % MATLAB structs must start with a letter, and can only contain
     % letters, digits, and underscores.
-    % Remove an initial slash.
-    col_label = regexprep(col_label, '^/', '');
-    % Replace '/' and '|' if they are present.
-    col_label = strrep(col_label,'/', '_');
-    col_label = strrep(col_label,'|', '_');
+    illegalChars = [{' '} {'*'} {'.'} {'/'} {'-'} {'|'}];
+    for i = 1 : length(illegalChars)
+        if ~isempty(strfind(col_label, illegalChars{i}))
+           col_label = strrep(col_label,illegalChars{i}, '_'); 
+           disp(['Illegal Column label. ' disp_label ' changed to ' col_label ]); 
```

Comment:
If there are multiple different illegal characters, the user will get this warning multiple times. For example, a column label `hi*/` would generate:

```
Illegal Column Label. hi*/ changed to hi_/
Illegal Column Label. hi_/ changed to hi__
```

Ideally the user would only see the warning once:

```
Illegal Column Label. hi*/ changed to hi__
```

You could use a boolean in the for-loop to track if any of the illegal characters are present, and check that boolean outside of the loop to determine if you should print a message.